adapters/IronSource/IronSourceAdapter/GADMAdapterIronSourceBannerAd.m

```diff
@@ -0,0 +1,135 @@
+
+// The completion handler to call when the ad loading succeeds or fails.
+@property(copy, nonatomic) GADMediationBannerLoadCompletionHandler adLoadCompletionHandler;
+
+// Ad configuration for the ad to be rendered.
+@property(weak, nonatomic) GADMediationAdConfiguration *adConfiguration;
+
+// An ad event delegate to invoke when ad rendering events occur.
+@property(weak, nonatomic) id<GADMediationBannerAdEventDelegate> adEventDelegate;
+
+/// Holds the ID of the ad instance to be presented.
+@property(nonatomic, copy) NSString *instanceID;
+
+/// Holds the state of the ad instance to be presented.
+@property(nonatomic, copy) NSString *instanceState;
+
+@property (nonatomic, strong) ISDemandOnlyBannerView *iSDemandOnlyBannerView;
+
+@end
+
+@implementation GADMAdapterIronSourceBannerAd
+
+- (instancetype)initWithGADMediationBannerConfiguration:(GADMediationBannerAdConfiguration *)adConfiguration
+                                      completionHandler:(GADMediationBannerLoadCompletionHandler)completionHandler {
+    self = [super init];
+    if (self) {
+        _adLoadCompletionHandler = completionHandler;
+        _adConfiguration = adConfiguration;
+        // Default instance ID
+        self.instanceID = GADMIronSourceDefaultInstanceId;
+        // Default instance state
+        self.instanceState = GADMAdapterIronSourceInstanceStateStart;
+    }
+    
+    return self;
+}
+
+- (void)renderBannerForAdConfig:(nonnull GADMediationBannerAdConfiguration *)adConfig
+              completionHandler:(nonnull GADMediationBannerLoadCompletionHandler)handler {
+    _adConfiguration = adConfig;
+    NSDictionary *credentials = [_adConfiguration.credentials settings];
+    
+    /* Parse application key */
+    NSString *applicationKey = @"";
+    if (credentials[GADMAdapterIronSourceAppKey]) {
+        applicationKey = credentials[GADMAdapterIronSourceAppKey];
+    }
+    
+    if ([GADMAdapterIronSourceUtils isEmpty:applicationKey]) {
+        NSError *error = GADMAdapterIronSourceErrorWithCodeAndDescription(
+                                                                          GADMAdapterIronSourceErrorInvalidServerParameters,
+                                                                          @"'appKey' parameter is missing. Make sure that appKey' server parameter is added.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    if (credentials[GADMAdapterIronSourceInstanceId]) {
+        self.instanceID = credentials[GADMAdapterIronSourceInstanceId];
+    }
+    
+    [[ISMediationManager sharedManager]
+     initIronSourceSDKWithAppKey:applicationKey
+     forAdUnits:[NSSet setWithObject:IS_BANNER]];
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
+        ISMediationManager *sharedManager = [ISMediationManager sharedManager];
+        [sharedManager initIronSourceSDKWithAppKey:applicationKey
+                                        forAdUnits:[NSSet setWithObject:IS_BANNER]];
```

Comment:
Any reason this is done twice? This was already called above?

On the other classes, this is not called in a `dispatch_async()` block as well. Can you confirm if this is needed or not?

Reply:
Fixed